fedn/cli/project_cmd.py

```diff
@@ -0,0 +1,220 @@
+            click.secho(f"No project with id '{id}' exists.", fg="red")
+        else:
+            print_response(response, "project", True)
+    else:
+        click.secho(f"Unexpected error: {response.status_code}", fg="red")
+
+
+@click.option("-id", "--id", required=True, help="id name of project.")
+@click.option("-p", "--protocol", required=False, default=STUDIO_DEFAULTS["protocol"], help="Communication protocol of studio (api)")
+@click.option("-H", "--host", required=False, default=STUDIO_DEFAULTS["host"], help="Hostname of studio (api)")
+@project_cmd.command("set-context")
+@click.pass_context
+def set_active_project(ctx, id: str = None, protocol: str = None, host: str = None):
+    """Set active project.
+
+    :param ctx:
+    :param id:
+    """
+    activate_project(id, protocol, host)
+
+
+def activate_project(id: str = None, protocol: str = None, host: str = None):
+    """Sets project with give ID as active by updating context file."""
+    studio_api = True
+    headers_projects = {}
+    context_path = os.path.join(home_dir, ".fedn")
+    context_data = get_context(context_path)
+
+    user_access_token = context_data.get("User tokens").get("access")
+
+    response_projects = get_response(
+        protocol=protocol, host=host, port=None, endpoint="projects", token=user_access_token, headers=headers_projects, usr_api=studio_api, usr_token=False
+    )
+    if response_projects.status_code == 200:
+        projects_response_json = response_projects.json()
+        if len(projects_response_json) > 0:
+            if id is None:
+                headers_projects["X-Project-Slug"] = projects_response_json[0].get("slug")
+                id = projects_response_json[0].get("slug")
+            else:
+                for i in projects_response_json:
+                    if i.get("slug") == id:
+                        headers_projects["X-Project-Slug"] = i.get("slug")
+
+            controller_url = f"{protocol}://{host}/{id}-fedn-reducer"
+
+            response_project_tokens = get_response(
+                protocol=protocol,
+                host=host,
+                port=None,
+                endpoint="admin-token",
+                token=user_access_token,
+                headers=headers_projects,
+                usr_api=studio_api,
+                usr_token=False,
+            )
+            if response_project_tokens.status_code == 200:
+                project_tokens = response_project_tokens.json()
+                context_data["Active project tokens"] = project_tokens
+                context_data["Active project id"] = id
+                context_data["Active project url"] = controller_url
+
+                set_context(context_path, context_data)
+
+                click.secho(f"Project with slug {id} is now active.", fg="green")
+            else:
+                click.secho(f"Unexpected error: {response_project_tokens.status_code}", fg="red")
+        else:
+            click.echo("Set current context.")
```

Comment:
"Set current context"? That's what the functions try to do. My suggestion was "No projects available to set current context."